lib/dialects/abstract/connection-manager.js

```diff
@@ -184,11 +184,28 @@ ConnectionManager.prototype.getConnection = function(options) {
   var self = this;
   options = options || {};
 
-  return new Promise(function (resolve, reject) {
-    self.pool.acquire(function(err, connection) {
-      if (err) return reject(err);
-      resolve(connection);
-    }, options.priority, options.type, options.useMaster);
+  return Promise.resolve().then(function () {
+    if (self.sequelize.options.databaseVersion === 0) {
```

Comment:
If several concurrent calls request a transaction yes. The other alternative is to block once the first call is sent, but it would make the code slightly more complicated. I don't really think its a big problem that call getversion `maxConnections` number of times :)